Tests/EndpointSpec.swift

```diff
@@ -72,8 +72,8 @@ final class EndpointSpec: QuickSpec {
 
         it("returns a nil urlRequest for an invalid URL") {
             let badEndpoint = Endpoint<Empty>(url: "some invalid URL", sampleResponseClosure: { .networkResponse(200, Data()) }, method: .get, task: .requestPlain, httpHeaderFields: nil)
-
-            expect(badEndpoint.urlRequest).to(beNil())
+            let urlRequest = try? badEndpoint.urlRequest()
```

Comment:
This is good – can we also add a test that makes sure the error thrown is correct?

Reply:
Great idea 👍 Thanks for the input